rpki-validator/src/main/java/net/ripe/rpki/validator3/rrdp/RrdpServiceImpl.java

```diff
@@ -253,6 +254,7 @@ private int processDownloadedSnapshot(RpkiRepository rpkiRepository, RpkiReposit
 
             return counter.get();
         } catch (IOException e) {
+            rrdpMetrics.update(notification.snapshotUri, ErrorCodes.RRDP_CORRUPTED_SNAPSHOT);
             throw new RrdpException("Couldn't read snapshot: ", e);
```

Comment:
Why don't we throw exception with the appropriate error code and update the metrics where we catch the exception? I think it would save some repetitive code.

Reply:
In some cases metrics were updated while we're not throwing error (L175, L168), so I'll keep it this way.